changes/views.py

```diff
@@ -72,3 +76,62 @@ def post(self, request, *args, **kwargs):
                               action='momconnect_nonloss_optout',
                               data={'reason': 'sms_failure'})
         return Response(status=status.HTTP_201_CREATED)
+
+
+class ReceiveAdminOptout(generics.GenericAPIView):
+    permission_classes = (IsAuthenticated,)
+
+    def post(self, request, *args, **kwargs):
+
+        admin_serializer = AdminOptoutSerializer(data=request.data)
+        if admin_serializer.is_valid():
+            identity_id = admin_serializer.validated_data["registrant_id"]
+        else:
+            return Response(admin_serializer.errors,
+                            status=status.HTTP_400_BAD_REQUEST)
+
+        sbm_client = StageBasedMessagingApiClient(
+            api_url=settings.STAGE_BASED_MESSAGING_URL,
+            auth_token=settings.STAGE_BASED_MESSAGING_TOKEN
+        )
+
+        active_subs = sbm_client.get_subscriptions(
+            {'identity': identity_id, 'active': True}
+        )["results"]
+
+        actions = set()
+        for sub in active_subs:
+            messageset = sbm_client.get_messageset(sub["messageset"])
+            if "nurseconnect" in messageset["short_name"]:
+                actions.add("nurse_optout")
+            elif "pmtct" in messageset["short_name"]:
+                actions.add("pmtct_nonloss_optout")
+            else:
+                actions.add("momconnect_nonloss_optout")
+
+        source = Source.objects.get(user=self.request.user)
+        request.data["source"] = source.id
+
+        data = {"changes": []}
+        for action in actions:
+            change = {
+                "registrant_id": identity_id,
+                "action": action,
+                "data": {"reason": "other"},
+                "source": source.id,
+            }
+            data["changes"].append(change)
+
+        serializer = MultipleChangeSerializer(data=data)
+
+        if serializer.is_valid():
+            change_objects = serializer.save()
+
+            result = {"results": []}
+            for change in change_objects:
+                result["results"].append(model_to_dict(change))
```

Comment:
We can probably just use the `serializer.data` as the results to send back to the user.